include/zephyr/net/ethernet.h

```diff
@@ -352,8 +352,11 @@ enum ethernet_if_types {
 	/** IEEE 802.3 Ethernet (default) */
 	L2_ETH_IF_TYPE_ETHERNET,
 
-	/** IEEE 802.11 Wi-Fi*/
+	/** Offloaded IEEE 802.11 Wi-Fi (default Wi-Fi) */
 	L2_ETH_IF_TYPE_WIFI,
+
+	/** Native IEEE 802.11 Wi-Fi */
+	L2_ETH_IF_TYPE_WIFI_NATIVE,
```

Comment:
After looking at the logic in commit `net: wifi: Add support to handle Wi-Fi native to Wi-Fi management` - is this separate `WIFI_NATIVE` type even needed? It doesn't seem to rely on the interface type, but rather have a nice fallback mechanism - if there's no "shared" implementation found, it falls back to offloaded one.

Reply:
Well, if we have a `type` to convey, then this `enum` is needed to convey. It can also be used as a check for Native stack to ensure given interface is `native`.

Reply:
I have moved the two commits over to the other PR, and I will remove the WIFI_NATIVE altogether.